src/bubble-chart/bubble-series.component.ts

```diff
@@ -0,0 +1,246 @@
+        [pointerEvents]="circle.value === 0 ? 'none': 'all'"
+        [data]="circle.value"
+        [classNames]="circle.classNames"
+        (select)="onClick($event, circle.label)"
+        (activate)="activateCircle(circle)"
+        (deactivate)="deactivateCircle(circle)"
+        ngx-tooltip
+        [tooltipPlacement]="'top'"
+        [tooltipType]="'tooltip'"
+        [tooltipTitle]="getTooltipText(circle)"
+      />
+    </svg:g>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger('animationState', [
+      transition('void => *', [
+        style({
+          opacity: 0,
+        }),
+        animate(250, style({opacity: 1}))
+      ])
+    ])
+  ]
+})
+export class BubbleSeriesComponent implements OnChanges {
+
+  @Input() data;
+  @Input() type = 'standard';
+  @Input() xScale;
+  @Input() yScale;
+  @Input() rScale;
+  @Input() colors;
+  @Input() scaleType;
+  @Input() visibleValue;
+  @Input() activeEntries: any[];
+
+  @Output() select = new EventEmitter();
+  @Output() activate = new EventEmitter();
+  @Output() deactivate = new EventEmitter();
+
+  areaPath: any;
+  circles: any[];
+
+  constructor(private location: Location) {
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.update();
+  }
+
+  update(): void {
+    this.circles = this.getCircles();
+  }
+
+  getCircles(): any[] {
+    const seriesName = this.data.name;
+    const pageUrl = this.location.path();
+
+    return this.data.series.map((d, i) => {
+      const y = d.y;
+      const x = d.x;
+      const r = d.r;
+
+      const radius = this.rScale(r);
+      const tooltipLabel = formatLabel(d.name);
+
+      if (y) {
+        let cx;
+        if (this.scaleType === 'time') {
+          cx = this.xScale(x);
+        } else if (this.scaleType === 'linear') {
+          cx = this.xScale(Number(x));
+        } else {
+          cx = this.xScale(x);
+        }
+
+        const cy = this.yScale(this.type === 'standard' ? y : d.d1);
+        const height = this.yScale.range()[0] - cy;
+
+        const opacity = 1;
+
+        const gradientId = 'grad' + id().toString();
+        const gradientFill = `url(${pageUrl}#${gradientId})`;
+
+        let color;
+        if (this.colors.scaleType === 'linear') {
+          if (this.type === 'standard') {
+            color = this.colors.getColor(y);
+          } else {
+            color = this.colors.getColor(d.d1);
+          }
+        } else {
+          color = this.colors.getColor(seriesName);
+        }
+
+        return {
+          x,
+          y,
+          r,
+          classNames: [`circle-data-${i}`],
+          value: y,
+          label: x,
+          cx,
+          cy,
+          radius,
+          height,
+          tooltipLabel,
+          color,
+          opacity,
+          seriesName,
+          barVisible: false,
+          gradientId,
+          gradientFill,
+          gradientStops: this.getGradientStops(color),
+          min: d.min,
```

Comment:
we don't have min and max props here. Lets remove these and all references to them.